backend/internal/api/service_handlers.go

```diff
@@ -222,14 +223,56 @@ func (h *ServiceHandler) getViewportNodes(c *gin.Context) {
 
 // parseVault triggers a vault parsing operation
 func (h *ServiceHandler) parseVault(c *gin.Context) {
-	// TODO: Implement when VaultService is available
-	c.JSON(http.StatusNotImplemented, gin.H{"error": "Vault parsing not yet implemented"})
+	// Use request context
+	ctx := c.Request.Context()
+
+	// Check if vault service is available
+	if h.vaultService == nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Vault service not configured"})
+		return
+	}
+
+	// Check if parse is already running
+	status, err := h.vaultService.GetParseStatus(ctx)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get parse status"})
+		return
+	}
+
+	if status.Status == "parsing" {
+		c.JSON(http.StatusConflict, gin.H{"error": "Parse already in progress"})
+		return
+	}
+
+	// Start parsing
+	history, err := h.vaultService.ParseAndIndexVault(ctx)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusOK, history)
 }
 
 // getParseStatus returns the status of the latest parse operation
 func (h *ServiceHandler) getParseStatus(c *gin.Context) {
-	// TODO: Implement when VaultService is available
-	c.JSON(http.StatusNotImplemented, gin.H{"error": "Parse status not yet implemented"})
+	// Use request context
+	ctx := c.Request.Context()
+
+	// Check if vault service is available
+	if h.vaultService == nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Vault service not configured"})
+		return
+	}
+
+	// Get parse status
+	status, err := h.vaultService.GetParseStatus(ctx)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get parse status"})
+		return
```

Comment:
TODO(CL): The generic 'Failed to get parse status' error loses context. Consider:
1. Distinguish between 'no parse has run yet' vs actual errors
2. Return 404 if no parse history exists
3. Include last successful parse info even if current one failed

Example enhanced error response:
```json
{
  "error": "No parse operation found",
  "last_successful_parse": {
    "completed_at": "2025-07-13T10:00:00Z",
    "nodes_count": 1500
  }
}
```